promql/value.go

```diff
@@ -193,7 +194,8 @@ type Sample struct {
 	F float64
 	H *histogram.FloatHistogram
 
-	Metric labels.Labels
+	Metric         labels.Labels
+	ShouldDropName bool
```

Comment:
I _think_ I may just need to refactor the `Vector` type to make it a struct instead of a `[]Sample`, and then move the flag to `Vector`. I will give it a try

Reply:
What if we just drop `__name__` altogether in cases like `rate(alertmanager_http_request_duration_seconds_count[1m]) or up{foo="bar"}` ?

Reply:
On second thoughts I don't think that is a good idea.

It sounds like `Vector` does need `ShouldDropName` on each series.  Is this the only place `Sample` is used?
If not, maybe `Sample` needs to split into two types?

Reply:
The only other place I found where `Sample` is used outside the `promql` package is [here](https://github.com/prometheus/prometheus/blob/main/web/api/v1/json_codec.go#L132-L137). Since it is a custom marshaler with handpicked fields, I don't think we need to add a new type.